paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/SnapshotReaderImpl.java

```diff
@@ -289,6 +283,12 @@ private List<DataSplit> generateSplits(
                                         .orElse(null)
                                 : null;
                 for (SplitGenerator.SplitGroup splitGroup : splitGroups) {
+                    DataSplit.Builder builder =
+                            DataSplit.builder()
+                                    .withSnapshot(snapshotId)
+                                    .withPartition(partition)
+                                    .withBucket(bucket)
+                                    .isStreaming(isStreaming);
                     List<DataFileMeta> dataFiles = splitGroup.files;
                     builder.withDataFiles(dataFiles);
                     if (splitGroup.rawConvertible) {
```

Comment:
It can also be modified this way. 
I think the current implementation is okay too, what do you think? @JingsongLi 

```
builder.rawFiles(
               splitGroup.rawConvertible
                       ? convertToRawFiles(partition, bucket, dataFiles)
                       : Collections.emptyList());
```

Reply:
Okay,I think your suggestion is better. I'll make some changes